### ISSUE TITLE
Issue during making on kernel 6.1.43

### ISSUE THREAD

**Issue by JPP1** [lwfinger/rtw89-BT#28]
I have a Rock 5b with kernel 6.1.43.
Compiling  for wi-fi (https://github.com/lwfinger/rtw89) has been fine. 
The Wi-Fi works.
But for the bluetooth (https://github.com/lwfinger/rtw89-BT) I got somes issues:

```
JMD@claud:/usr/local/src/rtw89-BT# sudo make
make -C /lib/modules/6.1.43-vendor-rk35xx/build M=/usr/local/src/rtw89-BT modules
make[1]: Entering directory '/usr/src/linux-headers-6.1.43-vendor-rk35xx'
  CC [M]  /usr/local/src/rtw89-BT/btmtk.o
/usr/local/src/rtw89-BT/btmtk.c: In function ‘btmtk_coredump_notify’:
/usr/local/src/rtw89-BT/btmtk.c:91:14: error: ‘HCI_DEVCOREDUMP_IDLE’ undeclared (first use in this function)
   91 |         case HCI_DEVCOREDUMP_IDLE:
      |              ^~~~~~~~~~~~~~~~~~~~
/usr/local/src/rtw89-BT/btmtk.c:91:14: note: each undeclared identifier is reported only once for each function it appears in
/usr/local/src/rtw89-BT/btmtk.c:94:14: error: ‘HCI_DEVCOREDUMP_ACTIVE’ undeclared (first use in this function)
   94 |         case HCI_DEVCOREDUMP_ACTIVE:
      |              ^~~~~~~~~~~~~~~~~~~~~~
/usr/local/src/rtw89-BT/btmtk.c:97:14: error: ‘HCI_DEVCOREDUMP_TIMEOUT’ undeclared (first use in this function); did you mean ‘HCI_DISCONN_TIMEOUT’?
   97 |         case HCI_DEVCOREDUMP_TIMEOUT:
      |              ^~~~~~~~~~~~~~~~~~~~~~~
      |              HCI_DISCONN_TIMEOUT
/usr/local/src/rtw89-BT/btmtk.c:98:14: error: ‘HCI_DEVCOREDUMP_ABORT’ undeclared (first use in this function)
   98 |         case HCI_DEVCOREDUMP_ABORT:
      |              ^~~~~~~~~~~~~~~~~~~~~
/usr/local/src/rtw89-BT/btmtk.c:99:14: error: ‘HCI_DEVCOREDUMP_DONE’ undeclared (first use in this function)
   99 |         case HCI_DEVCOREDUMP_DONE:
      |              ^~~~~~~~~~~~~~~~~~~~
/usr/local/src/rtw89-BT/btmtk.c: In function ‘btmtk_register_coredump’:
/usr/local/src/rtw89-BT/btmtk.c:362:31: error: ‘HCI_DEVCOREDUMP_IDLE’ undeclared (first use in this function)
  362 |         data->cd_info.state = HCI_DEVCOREDUMP_IDLE;
      |                               ^~~~~~~~~~~~~~~~~~~~
/usr/local/src/rtw89-BT/btmtk.c:365:16: error: implicit declaration of function ‘hci_devcd_register’; did you mean ‘hci_dev_reset’? [-Werror=implicit-function-declaration]
  365 |         return hci_devcd_register(hdev, btmtk_coredump, btmtk_coredump_hdr,
      |                ^~~~~~~~~~~~~~~~~~
      |                hci_dev_reset
/usr/local/src/rtw89-BT/btmtk.c: In function ‘btmtk_process_coredump’:
/usr/local/src/rtw89-BT/btmtk.c:379:14: error: ‘HCI_DEVCOREDUMP_IDLE’ undeclared (first use in this function)
  379 |         case HCI_DEVCOREDUMP_IDLE:
      |              ^~~~~~~~~~~~~~~~~~~~
/usr/local/src/rtw89-BT/btmtk.c:380:23: error: implicit declaration of function ‘hci_devcd_init’; did you mean ‘hci_sock_init’? [-Werror=implicit-function-declaration]
  380 |                 err = hci_devcd_init(hdev, MTK_COREDUMP_SIZE);
      |                       ^~~~~~~~~~~~~~
      |                       hci_sock_init
/usr/local/src/rtw89-BT/btmtk.c:386:44: error: ‘struct hci_dev’ has no member named ‘dump’
  386 |                 schedule_delayed_work(&hdev->dump.dump_timeout,
      |                                            ^~
/usr/local/src/rtw89-BT/btmtk.c:389:14: error: ‘HCI_DEVCOREDUMP_ACTIVE’ undeclared (first use in this function)
  389 |         case HCI_DEVCOREDUMP_ACTIVE:
      |              ^~~~~~~~~~~~~~~~~~~~~~
/usr/local/src/rtw89-BT/btmtk.c:391:23: error: implicit declaration of function ‘hci_devcd_append’; did you mean ‘hci_dev_open’? [-Werror=implicit-function-declaration]
  391 |                 err = hci_devcd_append(hdev, skb);
      |                       ^~~~~~~~~~~~~~~~
      |                       hci_dev_open
/usr/local/src/rtw89-BT/btmtk.c:402:33: error: implicit declaration of function ‘hci_devcd_complete’ [-Werror=implicit-function-declaration]
  402 |                                 hci_devcd_complete(hdev);
      |                                 ^~~~~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:250: /usr/local/src/rtw89-BT/btmtk.o] Error 1
make[1]: *** [Makefile:2012: /usr/local/src/rtw89-BT] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-6.1.43-vendor-rk35xx'
make: *** [Makefile:34: all] Error 2
```

what should I do ?

**Comment by JPP1** [comment]
Hi, I noticed that this error comes because the coredump.h file is missing in the /lib/modules/6.1.43-vendor-rk35xx/build/include/net/bluetooth folder.
I also saw the wifi driver finally doesn't work correctly. 
If I can list all SSID  around me, I cannot connect to one. Always password issue.
And I tried to put the Wifi as AP. I got another issue (wpa_supplicant) ...

I think I didn't recover the correct headers package (linux-headers-vendor-rk35xx) for kernel 6.1.43.
If somebody can tell me where recover the armbian headers for kernel 6.1.43 that could help me strongly ...
Thanks !

**Comment by aledomu** [comment]
Same here, tried everything and even downgraded to 5.10 legacy kernel. They say that version has the best hardware support but I call BS. Specifically Bluetooth did not work either.

Having encountered many non-hardware issues while using Armbian in my Orange Pi 5 Plus and really the inconsistency of the distro as a whole I'm fairly confident that this is an issue on the Armbian side. I would report this and other stuff, but seeing how much they hand-wave made up excuses for WiFi issues ("oh that's finicky hardware"), how they tell you to install `armbian-firmware-full` after a proven record of doing nothing **in their own forums** and how they do not have the kernel packages hooks for managing U-Boot correctly after uninstalling a kernel package I won't lose my time doing so and dealing with their insane sense of priorities. Hopefully Collabora gets their drivers upstreamed sooner than later and we can then enjoy a serious distro.
